src/components/scenes/WcConnectScene.tsx

```diff
@@ -70,23 +68,26 @@ export const WcConnectScene = (props: Props) => {
     }
   }
 
-  const handleRequestDapp = async (walletId: string) => {
-    try {
-      const dApp = await currencyWallets[walletId].otherMethods.wcInit({ uri })
-      const dAppName = String(dApp.peerMeta.name).split(' ')[0]
-      setDappDetails({
-        subTitleText: sprintf(lstrings.wc_confirm_subtitle, dAppName),
-        bodyTitleText: sprintf(lstrings.wc_confirm_body_title, dAppName),
-        dAppImage: <FastImage style={styles.currencyLogo} source={{ uri: dApp.peerMeta.icons[0] }} />
-      })
-    } catch (e: any) {
-      showError('Failed to connect, try again.')
-      console.error('wcInit error:', e)
-      navigation.navigate('wcConnections', {})
-    }
-  }
+  const handleRequestDapp = React.useCallback(
+    async (walletId: string) => {
+      try {
+        const dApp = await currencyWallets[walletId].otherMethods.wcInit({ uri })
+        const dAppName = String(dApp.peerMeta.name).split(' ')[0]
+        setDappDetails({
+          subTitleText: sprintf(lstrings.wc_confirm_subtitle, dAppName),
+          bodyTitleText: sprintf(lstrings.wc_confirm_body_title, dAppName),
+          dAppImage: <FastImage style={styles.currencyLogo} source={{ uri: dApp.peerMeta.icons[0] }} />
+        })
+      } catch (e: any) {
+        showError('Failed to connect, try again.')
+        console.error('wcInit error:', e)
+        navigation.navigate('wcConnections', {})
+      }
+    },
+    [currencyWallets, navigation, styles.currencyLogo, uri]
+  )
 
-  const showWalletListModal = () => {
+  const showWalletListModal = React.useCallback(() => {
```

Comment:
This function should be wrapped in our `useHandler` hook, *not* `useCallback`. That will keep its identity stable. It should also be named `handleWalletListModal`, since it's being passed to an `onPress` prop.